src/SIL.XForge.Scripture/ClientApp/src/xforge-common/i18n.service.ts

```diff
@@ -81,6 +81,29 @@ export class I18nService {
       }
       return s;
     },
+    npi: (d: Date, options) => {
+      const o = {
+        year: 'numeric',
+        month: '2-digit',
+        day: '2-digit',
+        hour: '2-digit',
+        minute: '2-digit',
+        hour12: false,
+        numberingSystem: 'deva',
+        ...options
+      };
```

Comment:
Change this:
``` suggestion
      } as const;
 ```
So the type will be inferred as this:
``` ts
const o: {
    readonly showTimeZone?: boolean;
    readonly year: "numeric";
    readonly month: "2-digit";
    readonly day: "2-digit";
    readonly hour: "2-digit";
    readonly minute: "2-digit";
    readonly hour12: false;
    readonly numberingSystem: "deva";
}
```
Instead of this:
``` ts
const o: {
    showTimeZone?: boolean;
    year: string;
    month: string;
    day: string;
    hour: string;
    minute: string;
    hour12: boolean;
    numberingSystem: string;
}
```
Then the `as any` isn't needed a couple lines later.

Reply:
Done. This is really interesting! Using an inline object avoids all this, but when you extract to a variable, suddenly it can't infer the type anymore?
<!-- Reviewable comment -ODsJ2zr6emGYB1I5QBT:bdg8rv9 -->
<!-- Sent from Reviewable.io -->

Reply:
It's not that it can't, it's that it almost always shouldn't, because it would be very inconvenient.

Imagine you wrote this:
``` ts
let o = { year: 'numeric' }; // infers type { year: string; } (type widening)
// If it inferred { year: "numeric"; } the next line would fail:
o.year = '2-digit';
```

Adding `as const` tells it not to widen the type.

What would actually be better than `as const` though would be specifying the type:
```
const o: Intl.DateTimeFormatOptions = {
```

If you get a type wrong after specifying the type, you get a helpful error on the same line as the mistake:
``` ts
const o: Intl.DateTimeFormatOptions = {
  year: 'tumeric', // Type '"tumeric"' is not assignable to type '"numeric" | "2-digit"'. Did you mean '"numeric"'?
  month: '2-digit',
  day: '2-digit',
  hour: '2-digit',
  minute: '2-digit',
  hour12: false,
  numberingSystem: 'deva',
  ...options
};
```

On the other hand, if you use `as const`, the problem isn't nearly as apparent until later, and the error is much less helpful:
``` ts
const o = {
  year: 'tumeric',
  month: '2-digit',
  day: '2-digit',
  hour: '2-digit',
  minute: '2-digit',
  hour12: false,
  numberingSystem: 'deva',
  ...options
} as const;

const parts = new Intl.DateTimeFormat('npi', o).formatToParts(d);
// No overload matches this call.
//   Overload 1 of 2, '(locales?: LocalesArgument, options?: DateTimeFormatOptions): DateTimeFormat', gave the following error.
//     Argument of type '{ readonly showTimeZone?: boolean; readonly year: "tumeric"; readonly month: "2-digit"; readonly day: "2-digit"; readonly hour: "2-digit"; readonly minute: "2-digit"; readonly hour12: false; readonly numberingSystem: "deva"; }' is not assignable to parameter of type 'DateTimeFormatOptions'.
//       Types of property 'year' are incompatible.
//         Type '"tumeric"' is not assignable to type '"numeric" | "2-digit"'. Did you mean '"numeric"'?
//   Overload 2 of 2, '(locales?: string | string[], options?: DateTimeFormatOptions): DateTimeFormat', gave the following error.
//     Argument of type '{ readonly showTimeZone?: boolean; readonly year: "tumeric"; readonly month: "2-digit"; readonly day: "2-digit"; readonly hour: "2-digit"; readonly minute: "2-digit"; readonly hour12: false; readonly numberingSystem: "deva"; }' is not assignable to parameter of type 'DateTimeFormatOptions'.ts(2769)